### PR TITLE
Fix update reactive keys for voltage control

### PR DESCRIPTION
**Pull request by caioluke**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

Bug fix

**What is the current behavior?**

Reactive keys for voltage control are not being updated correctly in SA after a generator is lost

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

### CODE REVIEW

File: src/test/java/com/powsybl/openloadflow/network/VoltageControlNetworkFactory.java [hunk 1]
```diff
@@ -1383,4 +1383,29 @@ public static Network createWithShuntSharedRemoteControl2() {
...
+        g1.setMaxP(10);
+        Generator g1Bis = network.getGenerator("g1Bis");
+        g1Bis.setMaxP(10);
+        Generator g4 = network.getGenerator("g4");
+        g4.setMaxP(20);
+        Terminal regTerminal = network.getLine("l34").getTerminal2();
+
+        g1.setRegulatingTerminal(regTerminal);
+        g1.setTargetV(1.6);
+        g1.setVoltageRegulatorOn(true);
```
Comment (jeandemanged) [inline src/test/java/com/powsybl/openloadflow/network/VoltageControlNetworkFactory.java]: style: chain these set calls, also together with the setMaxP above

File: src/test/java/com/powsybl/openloadflow/network/VoltageControlNetworkFactory.java [hunk 2]
```diff
@@ -1383,4 +1383,29 @@ public static Network createWithShuntSharedRemoteControl2() {
...
+        Generator g1 = network.getGenerator("g1");
+        g1.setMaxP(10);
+        Generator g1Bis = network.getGenerator("g1Bis");
+        g1Bis.setMaxP(10);
+        Generator g4 = network.getGenerator("g4");
+        g4.setMaxP(20);
+        Terminal regTerminal = network.getLine("l34").getTerminal2();
+
+        g1.setRegulatingTerminal(regTerminal);
+        g1.setTargetV(1.6);
```
Comment (jeandemanged) [inline src/test/java/com/powsybl/openloadflow/network/VoltageControlNetworkFactory.java]: functional: The 4 bus network has 1kV nominal voltage, why do you need such high unrealistic target for the purpose of reactive keys test ?

File: src/test/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisTest.java [hunk 3]
```diff
@@ -2024,6 +2027,47 @@ void testWithShuntAndGeneratorVoltageControls2() {
...
+        List<Contingency> contingencies = List.of(
+                new Contingency("g1", List.of(new GeneratorContingency("g1")))
+        );
+        LoadFlowParameters lfParameters = new LoadFlowParameters();
+        lfParameters.setUseReactiveLimits(true);
+        OpenLoadFlowParameters openLoadFlowParameters = OpenLoadFlowParameters.create(lfParameters);
+        openLoadFlowParameters.setReactivePowerRemoteControl(true)
+                .setMaxNewtonRaphsonIterations(30)
+                .setMaxRealisticVoltage(3)
+                .setStateVectorScalingMode(StateVectorScalingMode.LINE_SEARCH)
```
Comment (jeandemanged) [inline src/test/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisTest.java]: functional: should not need to do this, this isn't a unit test any more. Likely linked to 1.6kV targetV

**Review by jeandemanged — CHANGES_REQUESTED** [review]

File: src/test/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisTest.java [hunk 4]
```diff
@@ -2024,6 +2027,47 @@ void testWithShuntAndGeneratorVoltageControls2() {
         assertEquals(133.557, result.getPostContingencyResults().get(0).getNetworkResult().getBusResult("BUS_2").getV(), LoadFlowAssert.DELTA_V);
     }
 
+    @Test
+    void testWithSharedVoltageRemoteControl() {
+        Network network = VoltageControlNetworkFactory.createWithSharedRemoteControl2GensOnSameBusAnd1Extra();
```
[the remark this answers is not in the document]
Comment (caioluke) [reply src/test/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisTest.java:2031]: For no particular reason, I created this factory to test something in my other PR (the one about reactive power control) and then I reused it for this one too

**Review by jeandemanged — APPROVED** [review]

**Review by annetill — APPROVED** [review]

**Pull request merged**